components/ProjectList.js

```diff
@@ -2,29 +2,79 @@ import styled from "styled-components";
 import Link from "next/link";
 import { useRouter } from "next/router";
 import { useState } from "react";
+import dynamic from "next/dynamic";
+
+const ModalDeleteProject = dynamic(
+  () => import("@/components/ModalDeleteProject"),
+  {
+    ssr: false,
+  }
+);
 
 export default function ProjectList({ projects, handleDeleteProject }) {
+  const [searchTerm, setSearchTerm] = useState("");
+  const [projectToDelete, setProjectToDelete] = useState(null);
+
   return (
     <Wrapper>
       <StyledList>
-        {projects.map((project, index) => (
-          <ProjectItem key={index}>
-            <Link href={`/project/${project.id}`}>{project.title}</Link>
-            <DeleteButton
-              type="button"
-              onClick={() => handleDeleteProject(project.id)}
-            >
-              x
-            </DeleteButton>
-          </ProjectItem>
-        ))}
+        <SearchBar
+          type="text"
+          placeholder="search"
+          onChange={(event) => {
+            setSearchTerm(event.target.value);
+          }}
+        />
+        {projects
+          .filter((val) => {
+            if (searchTerm == "") {
+              return val;
+            } else if (
+              val.title.toLowerCase().includes(searchTerm.toLowerCase())
+            ) {
+              return val;
+            }
+          })
+          .map((project, index) => (
+            <ProjectItem key={index}>
+              <Link href={`/project/${project.id}`}>{project.title}</Link>
+              <DeleteButton
+                type="button"
+                onClick={() => setProjectToDelete(project.id)}
+              >
+                x
+              </DeleteButton>
+            </ProjectItem>
+          ))}
       </StyledList>
+      <ModalDeleteProject
+        appearModalDeleteProject={projectToDelete}
+        handleDeleteProject={() => handleDeleteProject(projectToDelete)}
+        handleClose={() => {
+          setProjectToDelete(null);
+        }}
+      />
     </Wrapper>
   );
 }
 
+const SearchBar = styled.input`
+  width: 90%;
+  margin: 5px;
+  padding: 10px;
+  border-radius: 7px;
+  border: 1px solid grey;
```

Comment:
i guess you will style your project in the next days but try to use responsive units consistently like em, rem

Reply:
yeah this is coming up next week ^^